packages/knip/src/plugins/relay/index.ts

```diff
@@ -0,0 +1,60 @@
+import parseArgs from 'minimist';
+import type { IsPluginEnabled, Plugin, ResolveEntryPaths } from '../../types/config.js';
+import { toDeferResolve, toEntry } from '../../util/input.js';
+import { isInternal, join } from '../../util/path.js';
+import { hasDependency } from '../../util/plugin.js';
+import type { RelayConfig } from './types.js';
+// https://relay.dev/docs/next/guides/compiler/#configuration
+// https://github.com/facebook/relay/blob/main/compiler/crates/relay-compiler/relay-compiler-config-schema.json
+
+const title = 'Relay';
+
+const enablers = ['vite-plugin-relay', '@swc/plugin-relay', 'babel-plugin-relay'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies, config }) =>
+  hasDependency(dependencies, enablers) || 'relay' in config;
+
+const config: string[] = ['relay.config.json', 'relay.config.js'];
+
+const resolveEntryPaths: ResolveEntryPaths<RelayConfig> = async config => {
+  const projects = 'projects' in config ? Object.values(config.projects) : [config];
+
+  return projects.flatMap(project => {
+    const artifactDirectory = project.artifactDirectory;
+
+    if (artifactDirectory == null) return [toEntry('**/__generated__/*')];
+
+    const inputs = [toEntry(join(artifactDirectory, '**'))];
+
+    if (project.requireCustomScalarTypes !== true) {
+      return inputs;
+    }
+
+    const scalars = Object.values(project.customScalarTypes ?? {}).flatMap(customScalarType =>
+      typeof customScalarType === 'object'
+        ? [
+            isInternal(customScalarType.path)
+              ? toEntry(join(artifactDirectory, customScalarType.path))
```

Comment:
I think `toEntry` should be `toProductionEntry` so they're also included in `--production` mode